tools/2024-06-gitops-feature-spec.md

```diff
@@ -0,0 +1,345 @@
+
+## Existing customer problem
+<!-- <Write this in first person. You basically want to summarize what “I” as a customer am trying to accomplish, why the current experience is a problem and the impact it has on me, my team, my work and or biz, etc…. i.e. “When I try to do x aspect of cloud native app development, I have the following challenges / issues….<details>. Those issues result in <negative impact those challenges / issues have on your work and or business.> -->
+As an SRE, I am expected to make adjustments to the configuration of the Kubernetes cluster and the Radius Environment as needed. Today, I adjust the cluster as needed through my GitOps toolset via patching solutions like [Kustomize](https://kubernetes.io/docs/tasks/manage-kubernetes-objects/kustomization/) in [Flux](https://fluxcd.io/flux/components/kustomize/ or [ArgoCD](https://argo-cd.readthedocs.io/en/stable/user-guide/kustomize/), but I'm unable to do so for all the Radius-specific resources since Radius types and Bicep files are not currently readable by Flux or ArgoCD. This means that I have to manually manage the Radius instance I am providing for my developers and operators to use, which is a significant overhead for me to adopt and to provide value to my development counterparts.
+
+## Desired customer experience outcome
+<!-- <Write this as an “I statement” that expresses the new capability from customer perspective … i.e. After this scenario is implemented “I can do, x, y, z, steps in cloud native app developer and seamlessly blah blah blah …. As a result <summarize positive impact on your work / business>  -->
+As an example, the application team will require 10 replica pods for their prod environment however this variable is currently set to 1. To fix this as a SRE, I should have 2 options:
+
+**Option 1**
+
+1. Using my editor of choice, VSCode, I'll make changes to the replica number for the pods through a YAML patch file that can contain the path to parameter annotations that effect the replica number of pods:
+
+Example:
+
+```yaml
+apiVersion: kustomize...
+kind: Component
+resources:
+- backend/mongoDB.yaml
+patches:
+- target:
+    group: core.opera.prod
+    version: 1
+    kind: Application
+    name: backend
+  path: backend/replica-patch.yaml
+```
+
+2. This file points to replica-patch.yaml which contains parameters to change the replica number of pods to 10.
+2. These 2 files are then committed to my configurational files to patch my Kubernetes cluster.
+
+> As an SRE, I'm very happy with this experience because I did not have to change anything in my normal workflow behavior and was able to leverage Flux and my normal YAML file workflow.
+
+> This is the current state for GitOps and Kustomize, thus should not require any new features to be implemented in Radius.
+
+**Option 2 Advanced SRE**
```

Comment:
I'm not sure this is advanced, I think it's table stakes.

Reply:
agreed.